src/components/azk/azk-hook.ts

```diff
@@ -1,82 +1,111 @@
+  [5, 8],
 ];
 
 const leftEdge = [0, 1, 3, 6];
 const rightEdge = [0, 2, 5, 9];
 const bottomEdge = [6, 7, 8, 9];
 
-export const isWinningMove = (selectedTileIndex: number, tiles: TileState[]) => {
+export const isWinningMove = (
+  selectedTileIndex: number,
+  tiles: TileState[]
+) => {
   // calculate the connected component of same-colored tiles containing the selected tile
   const reachableTiles = new Set<number>();
   const visit = (index: number) => {
-    if (tiles[index] === tiles[selectedTileIndex] && !reachableTiles.has(index)) {
+    if (
+      tiles[index] === tiles[selectedTileIndex] &&
+      !reachableTiles.has(index)
+    ) {
       reachableTiles.add(index);
       neighbors[index].forEach(neighbor => visit(neighbor));
     }
-  }
+  };
   visit(selectedTileIndex);
 
   // check if the reachable tiles include a tile from all three edges
-  const touchesEdge = (edge: number[]) => edge.some(tile => reachableTiles.has(tile));
-  return touchesEdge(leftEdge) && touchesEdge(rightEdge) && touchesEdge(bottomEdge);
-}
-
+  const touchesEdge = (edge: number[]) =>
+    edge.some(tile => reachableTiles.has(tile));
+  return (
+    touchesEdge(leftEdge) && touchesEdge(rightEdge) && touchesEdge(bottomEdge)
+  );
+};
 
 export const useAzkGame = (gameData: AzkGameData) => {
-  const [tiles, setTiles] = useState<TileState[]>(Array(10).fill("empty"));
-
-  const [playerOnTurn, setPlayerOnTurn] = useState<Player>("A");
+  const [tiles, setTiles] = useState<TileState[]>(Array(10).fill('empty'));
+  const [playerOnTurn, setPlayerOnTurn] = useState<Player>('A');
   const [winner, setWinner] = useState<Player | null>(null);
   const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
 
-  const currentQuestion = selectedIndex === null ? null : gameData.questions[selectedIndex].question;
+  const currentQuestion =
+    selectedIndex === null ? null : gameData.questions[selectedIndex].question;
 
-  const opponentOf = (player: Player) => player == "A" ? "B" : "A";
+  const opponentOf = (player: Player) => (player == 'A' ? 'B' : 'A');
+
+  const selectedTile = (selectedIndex: number) => {
+    const choosedTile = tiles.map((state, index) =>
+      index === selectedIndex ? 'choosed' : state
+    );
+    setTiles(choosedTile);
+  };
 
   const selectTile = async (index: number) => {
     if (index < 0 || index >= gameData.questions.length) {
-      throw new Error("Invalid tile index");
+      throw new Error('Invalid tile index');
     }
     if (winner) {
-      throw new Error("Game is already over");
+      throw new Error('Game is already over');
     }
     if (selectedIndex !== null) {
-      throw new Error("Tile already selected");
+      throw new Error('Tile already selected');
     }
-    if (tiles[index] !== "empty") {
-      throw new Error("Tile already taken");
+    if (tiles[index] !== 'empty') {
+      throw new Error('Tile already taken');
     }
+
     setSelectedIndex(index);
-  }
+    selectedTile(index)
```

Comment:
Informaci o vybranem policku ted mame ulozenou dvakrat, v `selectedIndex` a v `tiles`. Melo by to byt ulozene jen jednou, a ta druha promenna by se mela odvodit z te prvni.
Ja to pozdeji upravim, nemusime tim tohle PR blokovat